Framework/Algorithms/src/ConjoinWorkspaces.cpp

```diff
@@ -145,6 +133,57 @@ void ConjoinWorkspaces::checkForOverlap(const MatrixWorkspace &ws1,
   }
 }
 
+/**
+ * Conjoin two event workspaces together, including the history
+ *
+ * @param ws1:: The first workspace
+ * @param ws2:: The second workspace
+ * @return :: A new workspace containing the conjoined workspaces
+*/
+API::MatrixWorkspace_sptr
+ConjoinWorkspaces::conjoinEvents(const DataObjects::EventWorkspace &ws1,
+                                 const DataObjects::EventWorkspace &ws2) {
+  this->validateInputs(ws1, ws2, false);
+
+  // Check there is no overlap
+  if (this->getProperty("CheckOverlapping")) {
+    this->checkForOverlap(ws1, ws2, false);
+    m_overlapChecked = true;
+  }
+
+  // Both are event workspaces. Use the special method
+  auto output = this->execEvent();
+
+  // Copy the history from the original workspace
+  output->history().addHistory(ws1.getHistory());
+  return output;
+}
+
+/**
+* Conjoin two histogram workspaces together, including the history
+*
+* @param ws1:: The first workspace
+* @param ws2:: The second workspace
+* @return :: A new workspace containing the conjoined workspaces
+*/
+API::MatrixWorkspace_sptr
+ConjoinWorkspaces::conjoinHistograms(const API::MatrixWorkspace &ws1,
+                                     const API::MatrixWorkspace &ws2) {
+  // Check that the input workspaces meet the requirements for this algorithm
+  this->validateInputs(ws1, ws2);
+
+  if (this->getProperty("CheckOverlapping")) {
+    this->checkForOverlap(ws1, ws2, true);
+    m_overlapChecked = true;
```

Comment:
If we've got a variable called `m_overlapChecked`, it seems to me fairly sensible to make sure that it's set to `true` whenever `checkForOverlap` is run (and it doesn't come at a massive cost, right?). 

I'd say we either want to set it or get rid of the member completely (since not setting it would render it pointless). If `fixSpectrumNumbers` were ever to happen elsewhere, possibly in the future, it seems like a good check to have in place.